lib/src/features/authentication/domain/user_model.dart

```diff
@@ -0,0 +1,22 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'user_model.freezed.dart';
+part 'user_model.g.dart';
+
+enum AuthProvider { google, email, apple, facebook }
+
+@freezed
```

Comment:
freezed has built in copywith and it by far goes in the best manner possible with riverpod

Reply:
Yeh I understand, but for now it has ver less use case. So jus for simplicity , make it simple.

Reply:
ok as you wish,

Reply:
Hello @XxThunderBlastxX i have commented out the freezed package from pubspec and manually wrote all the fromJson, copyWith in the model class this time so i think it would be okay